### PR TITLE
Add RemoveProblemReducer test

### PR DESCRIPTION
**Pull request by T-Davis**
RemoveProblemReducer unit test
Part of [UC10](https://docs.google.com/document/d/1WONN5L97YWlA3-9ZdlM6vQcmBx36m-aGm5xcjb-t16w/edit#heading=h.k3l9rzap2hfx) Dismiss ProblemPage

### CODE REVIEW

**Review by nickmeinhold — CHANGES_REQUESTED** [review]
Awesome work mate!! 

Can we discuss our policy on unrelated changes in PRs - I'd love to get clear on an agreement that doesn't create too much overhead or become too burdensome but doesn't let us get too far down a path that ends in messy PRs without clear boundaries.  

In this case the xcconfig files I would love to have in a separate PR (or just discuss and leave them in this one) - I've seen the change on my end too and wouldn't mind pausing to understand where it came from if possible.  

Changing integration_test to SDK import - I know we've discussed it but I think we should try and remind each other to at least comment on unrelated changes so the PR includes the note on where the change came from.  

It's hypocritical of me given a lot of my PRs have included lots of unrelated stuff but I figure let's try and pull each other up, assuming we agree on a policy of (say) unrelated changes should be moved out where it makes sense or commented on when it doesn't.  

Thanks mate!

File: pubspec.yaml [hunk 1]
```diff
@@ -51,6 +51,7 @@ dependencies:
 
 dev_dependencies:
   integration_test:
```
Comment (T-Davis) [inline pubspec.yaml:53]: This dependency is now included in the Flutter SDK, thus doesn't need to be gotten from pub.dev

**Comment by T-Davis** [comment]
@nickmeinhold Great call out mate! Always want to be making better PRs.
I think the policy is as you expected (just not followed on my end) that the committed code should be relevant to the PR, and any code that isn't clearly relevant should be commented on to explain why it's relevant. 👍 
Thanks a ton! 🙏

File: test/sections/error-handling/UC10/remove_problem_reducer_unit_test.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import 'package:the_process/models/navigation/page_data/problem_page_data.dart';
+import 'package:the_process/models/problems/problem.dart';
+import 'package:the_process/reducers/problems/remove_problem.dart';
+
+void main() {
+  group('RemoveProblemReducer', () {
+    final problem = Problem(errorString: 'Problem error message');
+    final problemPageData = ProblemPageData(problem: problem);
+    test(
+        'Removes Problem from appState.problems and ProblemPageData from appState.pagesData',
```
Comment (nickmeinhold) [inline test/sections/error-handling/UC10/remove_problem_reducer_unit_test.dart:13]:
```suggestion
        'removes Problem from appState.problems and ProblemPageData from appState.pagesData',
```

Uber nit - the string in group('... is the start of the sentence and the string in test('... is the continuation of the sentence so should start lower case.

**Review by nickmeinhold — DISMISSED** [review]
Awesome work mate!  I just had the one super nit

**Review by nickmeinhold — APPROVED** [review]

**Comment by nickmeinhold** [comment]
@T-Davis I committed the nit and Approved my own change to save time... now that I type that out loud, that's probably not appropriate.  Let me know if you have any concerns.  Thanks mate!

**Pull request merged**